test/langtools/jdk/javadoc/doclet/testSnippetTag/TestSnippetTag.java

```diff
@@ -1849,6 +1849,119 @@ record TestCase(Snippet snippet, String expectedOutput) { }
     }
 
     @Test
+    public void testPositiveExternalHybridLangAttribute(Path base) throws Exception {
+
+        Path srcDir = base.resolve("src");
+        Path outDir = base.resolve("out");
+
+        record TestCase(String tag, String expectedContent, String expectedLang) { }
+
+        final var testCases = List.of(
+                // -------------------- external snippets --------------------
+                // if there's no file extension and no lang attribute, then
+                // markup is that of "java" and the class="language-" attribute
+                // is absent
+                new TestCase("""
+                        {@snippet file=".file"}
+                        """, """
+                        # @highlight substring=hi:
+                        hi <span class="bold">there</span>
```

Comment:
I see that these expected contents are duplicated in many places. Can we pull them out to local variables like `propertiesEvaluated` `javaEvaluated` and `noneEvaluated` for ease of future maintenance and clarity?